.github/workflows/build_publish.yaml

```diff
@@ -0,0 +1,137 @@
+# This workflow compiles the native wasmer_jni lib into various OS and architectures using dynamic_libs_reusable.yaml.
+# Then it builds and publishes the JAR based on the GITHUB_REF that calls the workflow. There are 2 options:
+# 1. The workflow is executed from a branch. In that case the JAR is available in the GitHub action as an artifact.
+# 2. The workflow is executed from a tag. In that case the JAR is published in the corresponding GitHub release.
+name: Build and publish
+
+on:
+  push:
+    branches:
+      - '*'    # Trigger on push to any branch
+    tags:
+      - '*'    # Trigger on push to any tag
+
+jobs:
+  upload_dynamic_libs:
+    strategy:
+      matrix:
+        platform:
+          - os: 'macos-latest'
+            target: 'aarch64-apple-darwin'
+            artifact: 'darwin-arm64'
+            lib_name: 'libwasmer_jni.dylib'
+
+          - os: 'macos-latest'
+            target: 'x86_64-apple-darwin'
+            artifact: 'darwin-amd64'
+            lib_name: 'libwasmer_jni.dylib'
+
+          - os: 'ubuntu-latest'
+            target: 'x86_64-unknown-linux-gnu'
+            artifact: 'linux-amd64'
+            lib_name: 'libwasmer_jni.so'
+
+          - os: 'windows-latest'
+            target: 'x86_64-pc-windows-msvc'
+            artifact: 'win32-amd64'
+            lib_name: 'wasmer_jni.dll'
+    uses: ./.github/workflows/dynamic_libs_reusable.yaml
+    with:
+      platform_os: ${{ matrix.platform.os }}
+      platform_target: ${{ matrix.platform.target }}
+      platform_artifact: ${{ matrix.platform.artifact }}
+      platform_lib_name: ${{ matrix.platform.lib_name }}
+
+  publish_jar:
+    name: Publish the JARs
+    runs-on: ubuntu-latest
+    needs: [upload_dynamic_libs]
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set up Java 21
+        uses: actions/setup-java@v1
+        with:
+          java-version: 21
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          default: true
+          override: true
+
+      - name: Cache Cargo registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache Cargo bin
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/bin
+          key: cargo-bin-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache Cargo build
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Setup wasmer-jni artifacts dir
+        id: setip_dir
+        shell: bash
+        run: |
+          random_string=$(uuidgen)
```

Comment:
why not use a custom name like "dynamic_libs" or something similar?

Reply:
I think `mktemp` is the exact GNU cureutil for this job. [SO answer](https://stackoverflow.com/a/4632032)
I'm not sure whether that's necessary in a GH runner, but don't forget to delete it afterwards, too. 😄

Reply:
The VM used for each workflow run is destroyed and re-created from a fresh instance each run. Deleting isn't needed. Will make the change.